src/playlist-controller.js

```diff
@@ -929,6 +929,31 @@ export class PlaylistController extends videojs.EventTarget {
       this.onEndOfStream();
     });
 
+    // In DASH, there is the possibility of the video segment and the audio segment
+    // at a current time to be on different timelines. When this occurs, the player
+    // forwards playback to a point where these two segment types are back on the same
+    // timeline. This time will be just after the end of the audio segment that is on
+    // a previous timeline.
+    this.timelineChangeController_.on('audioTimelineBehind', () => {
+      const mainPendingTimelineChange = this.timelineChangeController_.pendingTimelineChange({ type: 'main' });
+
+      // Adjust the pending audio timeline to match the main timeline
+      this.timelineChangeController_.pendingTimelineChange({ to: mainPendingTimelineChange.to, from: mainPendingTimelineChange.from, type: 'audio' });
+
+      const segmentInfo = this.audioSegmentLoader_.pendingSegment_;
+
+      if (!segmentInfo || !segmentInfo.segment || !segmentInfo.segment.syncInfo) {
+        return;
+      }
+
+      // Update the current time to just after the faulty audio segment.
+      // This moves playback to a spot where both audio and video segments
+      // are on the same timeline.
+      const newTime = segmentInfo.segment.syncInfo.end + 0.01;
+
+      this.tech_.setCurrentTime(newTime);
```

Comment:
We buffer segments ahead of time. This will jump ahead of what a user sees on the screen at the moment of the jump. Is that expected?

Reply:
In this scenario, I believe what the user will see at the moment of the jump is a loading screen. This is because the segments that _should_ be playing are not loaded to due this timeline inconsistency.

In this case, I think the jump is OK, but let me know if I am off there.

Reply:
> This is because the segments that should be playing are not loaded to due this timeline inconsistency.

Yeah, that is what I am trying to say; the segment you are loading is around `bufferedEnd`, while what users see is around `currentTime`.  We do not load segments that should be playing right now, in most cases, we are loading segments future segments (ahead of time).

we load segments around currentTime only when we are switching rendition.

Reply:
I guess we should probably check whether currentTime === bufferedEnd, and we are really stuck so we can jump ahead. I suspect we should already have this logic in the playback watcher.